app/src/main/java/com/example/tasky/feature_agenda/data/repository/AgendaRepositoryImpl.kt

```diff
@@ -65,59 +87,43 @@ class AgendaRepositoryImpl(
             remoteEvents + remoteTasks + remoteReminders
 
         } catch (e: HttpException) {
-            return AgendaResult.Error(e.message ?: "Invalid Response")
+            return Result.Error(e.message ?: "Invalid Response")
         } catch (e: IOException) {
-            return AgendaResult.Error(e.message ?: "Couldn't reach server")
+            return Result.Error(e.message ?: "Couldn't reach server")
         }
 
-        remoteAgenda.let { agenda ->
-            db.eventDao.deleteEvents()
-            db.eventDao.deleteAttendees()
-            db.reminderDao.deleteReminders()
-            db.taskDao.deleteTasks()
+        db.eventDao.deleteEventsForSpecificDay(startOfDay, endOfDay)
+        db.eventDao.deleteAttendeesForSpecificDay(startOfDay, endOfDay)
+        db.reminderDao.deleteRemindersForSpecificDay(startOfDay, endOfDay)
+        db.taskDao.deleteTasksForSpecificDay(startOfDay, endOfDay)
 
-            val eventList = agenda.filterIsInstance<AgendaItem.Event>()
-            val remindersList = agenda.filterIsInstance<AgendaItem.Reminder>()
-            val tasksList = agenda.filterIsInstance<AgendaItem.Task>()
+            val eventList = remoteAgenda.filterIsInstance<AgendaItem.Event>()
+            val remindersList = remoteAgenda.filterIsInstance<AgendaItem.Reminder>()
+            val tasksList = remoteAgenda.filterIsInstance<AgendaItem.Task>()
 
             eventList.forEach { event->
-                db.eventDao.upsertEvent(event.toEventEntity())
-                db.eventDao.upsertAttendee(event.attendees.map { it.toAttendeeEntity(event.eventId) })
+                val eventEntity = event.toEventEntity()
+                val attendeeEntities = event.attendees.map { it.toAttendeeEntity(event.eventId) }
+
+                db.eventDao.upsertEvent(eventEntity)
+                db.eventDao.upsertAttendees(attendeeEntities)
             }
 
-            tasksList.forEach {
-                db.taskDao.upsertTask(it.toTaskEntity())
+            tasksList.forEach { task ->
+                val taskEntity = task.toTaskEntity()
+                db.taskDao.upsertTask(taskEntity)
             }
 
-            remindersList.forEach {
-                db.reminderDao.upsertReminder(it.toReminderEntity())
+            remindersList.forEach { reminder ->
+                val reminderEntity = reminder.toReminderEntity()
+                db.reminderDao.upsertReminder(reminderEntity)
             }
```

Comment:
Here, I would recommend to use a database transaction instead. That will ensure that all these DB calls will only be applied to the DB if they all succeed. If one of these wouldn't succeed for whatever reason (out of storage, coroutine gets cancelled, etc.), your DB will be in an inconsistent state. The first calls may have already succeeded, but the later calls will never be executed due to a cancellation for example.
In Room, you can create a transaction like this for example:
```
@Dao
interface AgendaDao {

    @Transaction
    suspend fun updateAgenda(db: AgendaDatabase, agendaItems: List<AgendaItem>) {
        db.apply {
            eventDao.deleteAllEvents()
            attendeeDao.deleteAllAttendees()
            taskDao.deleteAllTasks()
            reminderDao.deleteAllReminders()

            val events = agendaItems.filterIsInstance<AgendaItem.Event>()
            val tasks = agendaItems.filterIsInstance<AgendaItem.Task>()
            val reminders = agendaItems.filterIsInstance<AgendaItem.Reminder>()
            events.forEach {
                eventDao.insertEvent(it.toEventEntity())
                it.attendees.forEach {
                    attendeeDao.insertAttendee(it.toAttendeeEntity)
                }
            }
            tasks.forEach {
                taskDao.insertTask(it.toTaskEntity())
            }
            reminders.forEach {
                reminderDao.insertReminder(it.toReminderEntity())
            }
        }
    }
}
```
Whenever you need to execute multiple DB calls that should only be applied to the DB if they all succeed, you need something like this.